proc/proc.go

```diff
@@ -580,21 +594,26 @@ func initializeDebugProcess(dbp *Process, path string, attach bool) (*Process, e
 		return nil, err
 	}
 
-	if err := dbp.updateThreadList(); err != nil {
-		return nil, err
-	}
-
 	switch runtime.GOARCH {
 	case "amd64":
 		dbp.arch = AMD64Arch()
 	}
 
+	if err := dbp.updateThreadList(); err != nil {
+		return nil, err
+	}
+
 	ver, isextld, err := dbp.getGoInformation()
 	if err != nil {
 		return nil, err
 	}
 
 	dbp.arch.SetGStructOffset(ver, isextld)
+	// SelectedGoroutine can not be set correctly by the call to updateThreadList
+	// because without calling SetGStructOffset we can not read the G struct of CurrentThread
+	// but without calling updateThreadList we can not examine memory to determine
+	// the offset of g struct inside TLS
+	dbp.SwitchThread(dbp.CurrentThread.Id)
```

Comment:
Instead of calling `SwitchThread` for the side effect of setting `SelectedGoroutine` let's just set it as such:

`dbp.SelectedGoroutine, err = dbp.CurrentThread.GetG()`

That makes the intention of the code more clear and at that point we can probably do without the comments.